src/translate.py

```diff
@@ -1495,20 +1521,25 @@ def __contains__(self, key: Register) -> bool:
 
     def __setitem__(self, key: Register, value: Expression) -> None:
         assert key != Register("zero")
-        self.contents[key] = value
+        self.contents[key] = RegData(value, RegDataMeta())
+
+    def set_raw(self, key: Register, data: RegData) -> None:
+        assert key != Register("zero")
+        self.contents[key] = data
```

Comment:
This feels asymmetrical? This is the only function which takes (or returns) a `RegData`, it's not the "complement" to `get_raw(...) -> Optional[Expression]`?

I think I'd either rename this to something that's distinct from `get_raw()`, like `set_regdata()`? Or refactor `get_raw()` to return an `Optional[RegData]` (but that sounds harder).

Reply:
There is also a lot of code which accesses `RegInfo.contents` directly, like `info.contents.get(reg)`. 
Do you think this is worth encapsulating?

Reply:
Added `get_meta` and `set_with_meta` accessors and removed `set_raw`. There's still some two places that access `contents` directly but it feels better now at least.